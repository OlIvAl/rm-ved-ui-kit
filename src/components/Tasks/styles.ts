import { makeStyles } from "@material-ui/core/styles";

export const useStyles = makeStyles({
  paper: {
    display: "flex",
    flexDirection: "column",
    paddingRight: 0,
    width: 468,
    height: 800
  },
  paperHeader: {
    paddingRight: 24
  },
  subtitle: {
    marginBottom: 24
  },
  select: {
    width: "100%"
  },
  divider: {
    margin: "16px 0"
  },
  taskWrapper: {
    paddingRight: 24,
    overflow: "auto",
    maxHeight: "708px"
  },
  task: {
    padding: "8px 16px",
    borderRadius: 4,
    margin: "16px 0",
    "&:last-child, &:first-child": {
      margin: 0
    }
  },
  expansion: {
    background: "#c3ffd8"
  },
  attract: {
    background: "#f9e0ff"
  },
  service: {
    background: "#e6eeff"
  },
  taskContent: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center"
  },
  substituted: {
    display: "flex",
    width: 54,
    height: 20,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 10,
    background: "#fff",
    marginRight: 16
  },
  substitutedIcon: {
    marginRight: 4
  },
  trigger: { flex: 1 }
});
