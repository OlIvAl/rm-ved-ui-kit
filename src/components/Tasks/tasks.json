[
  {
    "id": "as8d6f89a6sf",
    "companyName": "ООО Ромашка",
    "step": "First",
    "trigger": "Наименование триггера",
    "group": "attract",
    "substituted": true,
    "startDeadline": 1588865304
  },
  {
    "id": "as8d6f8934sf",
    "companyName": "ООО Ромашка",
    "step": "First",
    "trigger": "Наименование триггера",
    "group": "service",
    "substituted": true,
    "startDeadline": 1591650000
  },
  {
    "id": "as8d6f8566sf",
    "companyName": "ООО Ромашка",
    "step": "First",
    "trigger": "Наименование триггера",
    "group": "expansion",
    "substituted": true,
    "startDeadline": 1599598800
  },
  {
    "id": "as8d6789a6sf",
    "companyName": "ООО Ромашка",
    "step": "First",
    "trigger": "Наименование триггера",
    "group": "expansion",
    "substituted": true,
    "startDeadline": 1599598800
  },
  {
    "id": "as8d6f89444f",
    "companyName": "ООО Ромашка",
    "step": "First",
    "trigger": "Наименование триггера",
    "group": "attract",
    "substituted": true,
    "startDeadline": 1588149769
  },
  {
    "id": "a2424224a6sf",
    "companyName": "ООО Ромашка",
    "step": "First",
    "trigger": "Наименование триггера",
    "group": "expansion",
    "substituted": true,
    "startDeadline": 1588149769
  },
  {
    "id": "s3fdf9af0a9",
    "companyName": "ООО Ромашка",
    "step": "First",
    "trigger": "Наименование триггера",
    "group": "expansion",
    "substituted": true,
    "startDeadline": 1588149769
  },
  {
    "id": "as8d6f8944_f",
    "companyName": "ООО Ромашка",
    "step": "First",
    "trigger": "Наименование триггера",
    "group": "service",
    "substituted": true,
    "startDeadline": 1588149769
  },
  {
    "id": "a24242_4a6sf",
    "companyName": "ООО Ромашка",
    "step": "First",
    "trigger": "Наименование триггера",
    "group": "expansion",
    "substituted": true,
    "startDeadline": 1588149769
  },
  {
    "id": "s3_df9af0a9",
    "companyName": "ООО Ромашка",
    "step": "First",
    "trigger": "Наименование триггера",
    "group": "attract",
    "substituted": true,
    "startDeadline": 1588149769
  }
]
