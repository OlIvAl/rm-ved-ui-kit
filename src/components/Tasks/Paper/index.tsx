import MuiPaper from "@material-ui/core/Paper";
import { withStyles } from "@material-ui/core/styles";

const Paper = withStyles({
  root: {
    borderRadius: 12,
    padding: 24
  }
})(MuiPaper);

export default Paper;
