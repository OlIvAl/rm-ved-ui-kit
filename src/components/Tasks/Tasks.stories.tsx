import React from "react";
import { Tasks } from "./Tasks";
import TASKS from "./tasks.json";

export default {
  title: "Tasks"
};

export const Tasks_ = () => <Tasks tasks={TASKS as ITask[]} />;
