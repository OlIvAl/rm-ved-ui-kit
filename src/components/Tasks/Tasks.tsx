import React, { FC } from "react";
import Paper from "./Paper";
import Subtitle from "./Titles/Subtitle";
import MenuItem from "@material-ui/core/MenuItem";
import { Select, Typography } from "@material-ui/core";
import Divider from "@material-ui/core/Divider";
import Box from "@material-ui/core/Box";
import Card from "@material-ui/core/Card";
import { useStyles } from "./styles";
import SubstitutedIcon from "./icons/SubstitutedIcon";
import moment from "moment";

export default {
  title: "Tasks"
};

const getDayCounts = (date: number) => {
  const today = moment();
  const deadline = moment.unix(date);
  const diff = moment(deadline).diff(today, "days");
  return diff >= 0 ? diff + 1 : 0;
};

export const Tasks: FC<IProps> = ({ tasks }) => {
  const classes = useStyles();
  const [selectValue, setSelectValue] = React.useState<string>("Все задачи");

  const handleSecondSelectChange = (
    event: React.ChangeEvent<{ value: unknown }>
  ) => {
    setSelectValue(event.target.value as string);
  };

  return (
    <Paper className={classes.paper} variant="outlined">
      <div className={classes.paperHeader}>
        <Subtitle className={classes.subtitle}>
          Задачи ({tasks.length})
        </Subtitle>
        <Select
          value={selectValue}
          onChange={handleSecondSelectChange}
          variant="outlined"
          className={classes.select}
        >
          <MenuItem value="Все задачи">Все задачи</MenuItem>
          <MenuItem value="Привлечение">
            Привлечение&nbsp;
            <Box component="span" fontWeight="fontWeightMedium">
              (10)
            </Box>
          </MenuItem>
          <MenuItem value="Расширение">
            Расширение&nbsp;
            <Box component="span" fontWeight="fontWeightMedium">
              (2)
            </Box>
          </MenuItem>
          <MenuItem value="Сервисные">
            Сервисные&nbsp;
            <Box component="span" fontWeight="fontWeightMedium">
              (6)
            </Box>
          </MenuItem>
          <MenuItem value="Задачи по замещению">
            Задачи по замещению&nbsp;
            <Box component="span" fontWeight="fontWeightMedium">
              (1)
            </Box>
          </MenuItem>
        </Select>
        <Divider className={classes.divider} />
      </div>
      <div className={classes.taskWrapper}>
        {tasks.map((task: ITask) => (
          <Card
            key={task.id}
            className={`${classes.task} ${classes[task.group]}`}
            elevation={0}
          >
            <div>
              <Typography variant="subtitle1">{task.companyName}</Typography>
            </div>
            <div className={classes.taskContent}>
              <Typography className={classes.trigger}>
                {task.trigger}
              </Typography>
              {task.substituted && (
                <div className={classes.substituted}>
                  <SubstitutedIcon className={classes.substitutedIcon} />
                  <Typography>зам</Typography>
                </div>
              )}
              <Typography
                variant="subtitle1"
                {...(getDayCounts(task.startDeadline)
                  ? { color: "textPrimary" }
                  : { color: "error" })}
              >
                {getDayCounts(task.startDeadline)}
              </Typography>
            </div>
          </Card>
        ))}
      </div>
    </Paper>
  );
};
