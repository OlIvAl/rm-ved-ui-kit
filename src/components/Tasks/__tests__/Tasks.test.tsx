import React, { ReactElement } from "react";
import { render } from "@testing-library/react";
import { StylesProvider, StylesOptions } from "@material-ui/styles/";
import { Tasks } from "../Tasks";
import TASKS from "../tasks.json";

const generateClassName: StylesOptions["generateClassName"] = (
  rule,
  sheet
): string => `${sheet!.options.classNamePrefix}-${rule.key}`;

const provideTheme = (ui: ReactElement): ReactElement => {
  return (
    <StylesProvider generateClassName={generateClassName}>{ui}</StylesProvider>
  );
};

describe("Tasks", () => {
  it("render Tasks component with tasks array", done => {
    const { container } = render(
      provideTheme(<Tasks tasks={TASKS as ITask[]} />)
    );

    expect(container).toMatchSnapshot();
    done();
  });

  it("render Tasks component with one task", done => {
    const { container } = render(
      provideTheme(
        <Tasks
          tasks={[
            {
              id: "a2424224a6sf",
              companyName: "ООО Ромашка",
              step: "First",
              trigger: "Наименование триггера",
              group: "expansion" as any,
              substituted: false,
              startDeadline: 1588149769
            }
          ]}
        />
      )
    );

    expect(container).toMatchSnapshot();
    done();
  });

  it("render Tasks component with empty array", done => {
    const { container } = render(provideTheme(<Tasks tasks={[]} />));

    expect(container).toMatchSnapshot();
    done();
  });
});
