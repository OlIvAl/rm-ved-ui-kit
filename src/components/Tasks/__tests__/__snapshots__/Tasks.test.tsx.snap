// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks render Tasks component with empty array 1`] = `
<div>
  <div
    class="MuiPaper-root WithStyles(ForwardRef(Paper))-root makeStyles-paper MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      class="makeStyles-paperHeader"
    >
      <h6
        class="MuiTypography-root makeStyles-subtitle MuiTypography-subtitle1"
      >
        Задачи (
        0
        )
      </h6>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root makeStyles-select"
      >
        <div
          aria-haspopup="listbox"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          Все задачи
        </div>
        <input
          type="hidden"
          value="Все задачи"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <hr
        class="MuiDivider-root makeStyles-divider"
      />
    </div>
    <div
      class="makeStyles-taskWrapper"
    />
  </div>
</div>
`;

exports[`Tasks render Tasks component with one task 1`] = `
<div>
  <div
    class="MuiPaper-root WithStyles(ForwardRef(Paper))-root makeStyles-paper MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      class="makeStyles-paperHeader"
    >
      <h6
        class="MuiTypography-root makeStyles-subtitle MuiTypography-subtitle1"
      >
        Задачи (
        1
        )
      </h6>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root makeStyles-select"
      >
        <div
          aria-haspopup="listbox"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          Все задачи
        </div>
        <input
          type="hidden"
          value="Все задачи"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <hr
        class="MuiDivider-root makeStyles-divider"
      />
    </div>
    <div
      class="makeStyles-taskWrapper"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-task makeStyles-expansion MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            ООО Ромашка
          </h6>
        </div>
        <div
          class="makeStyles-taskContent"
        >
          <p
            class="MuiTypography-root makeStyles-trigger MuiTypography-body1"
          >
            Наименование триггера
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorError"
          >
            0
          </h6>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tasks render Tasks component with tasks array 1`] = `
<div>
  <div
    class="MuiPaper-root WithStyles(ForwardRef(Paper))-root makeStyles-paper MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      class="makeStyles-paperHeader"
    >
      <h6
        class="MuiTypography-root makeStyles-subtitle MuiTypography-subtitle1"
      >
        Задачи (
        10
        )
      </h6>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root makeStyles-select"
      >
        <div
          aria-haspopup="listbox"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          Все задачи
        </div>
        <input
          type="hidden"
          value="Все задачи"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <hr
        class="MuiDivider-root makeStyles-divider"
      />
    </div>
    <div
      class="makeStyles-taskWrapper"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-task makeStyles-attract MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            ООО Ромашка
          </h6>
        </div>
        <div
          class="makeStyles-taskContent"
        >
          <p
            class="MuiTypography-root makeStyles-trigger MuiTypography-body1"
          >
            Наименование триггера
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorError"
          >
            0
          </h6>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-task makeStyles-service MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            ООО Ромашка
          </h6>
        </div>
        <div
          class="makeStyles-taskContent"
        >
          <p
            class="MuiTypography-root makeStyles-trigger MuiTypography-body1"
          >
            Наименование триггера
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
          >
            28
          </h6>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-task makeStyles-expansion MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            ООО Ромашка
          </h6>
        </div>
        <div
          class="makeStyles-taskContent"
        >
          <p
            class="MuiTypography-root makeStyles-trigger MuiTypography-body1"
          >
            Наименование триггера
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
          >
            120
          </h6>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-task makeStyles-expansion MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            ООО Ромашка
          </h6>
        </div>
        <div
          class="makeStyles-taskContent"
        >
          <p
            class="MuiTypography-root makeStyles-trigger MuiTypography-body1"
          >
            Наименование триггера
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
          >
            120
          </h6>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-task makeStyles-attract MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            ООО Ромашка
          </h6>
        </div>
        <div
          class="makeStyles-taskContent"
        >
          <p
            class="MuiTypography-root makeStyles-trigger MuiTypography-body1"
          >
            Наименование триггера
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorError"
          >
            0
          </h6>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-task makeStyles-expansion MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            ООО Ромашка
          </h6>
        </div>
        <div
          class="makeStyles-taskContent"
        >
          <p
            class="MuiTypography-root makeStyles-trigger MuiTypography-body1"
          >
            Наименование триггера
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorError"
          >
            0
          </h6>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-task makeStyles-expansion MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            ООО Ромашка
          </h6>
        </div>
        <div
          class="makeStyles-taskContent"
        >
          <p
            class="MuiTypography-root makeStyles-trigger MuiTypography-body1"
          >
            Наименование триггера
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorError"
          >
            0
          </h6>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-task makeStyles-service MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            ООО Ромашка
          </h6>
        </div>
        <div
          class="makeStyles-taskContent"
        >
          <p
            class="MuiTypography-root makeStyles-trigger MuiTypography-body1"
          >
            Наименование триггера
          </p>
          <div
            class="makeStyles-substituted"
          >
            <svg
              class="makeStyles-substitutedIcon"
              height="12"
              width="8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6c2.1524944 0 3.907886 2.5502991 3.9964909 5.7463716L8 12H0l.0035091-.2536284C.0921141 8.5502991 1.8475056 6 4 6zm0-6c1.6568543 0 3 1.3431457 3 3S5.6568543 6 4 6C2.3431458 6 1 4.6568543 1 3s1.3431458-3 3-3z"
                fill="#3A3B3D"
                fill-rule="evenodd"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              зам
            </p>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorError"
          >
            0
          </h6>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-task makeStyles-expansion MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            ООО Ромашка
          </h6>
        </div>
        <div
          class="makeStyles-taskContent"
        >
          <p
            class="MuiTypography-root makeStyles-trigger MuiTypography-body1"
          >
            Наименование триггера
          </p>
          <div
            class="makeStyles-substituted"
          >
            <svg
              class="makeStyles-substitutedIcon"
              height="12"
              width="8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6c2.1524944 0 3.907886 2.5502991 3.9964909 5.7463716L8 12H0l.0035091-.2536284C.0921141 8.5502991 1.8475056 6 4 6zm0-6c1.6568543 0 3 1.3431457 3 3S5.6568543 6 4 6C2.3431458 6 1 4.6568543 1 3s1.3431458-3 3-3z"
                fill="#3A3B3D"
                fill-rule="evenodd"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              зам
            </p>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorError"
          >
            0
          </h6>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-task makeStyles-attract MuiPaper-elevation0 MuiPaper-rounded"
      >
        <div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            ООО Ромашка
          </h6>
        </div>
        <div
          class="makeStyles-taskContent"
        >
          <p
            class="MuiTypography-root makeStyles-trigger MuiTypography-body1"
          >
            Наименование триггера
          </p>
          <div
            class="makeStyles-substituted"
          >
            <svg
              class="makeStyles-substitutedIcon"
              height="12"
              width="8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6c2.1524944 0 3.907886 2.5502991 3.9964909 5.7463716L8 12H0l.0035091-.2536284C.0921141 8.5502991 1.8475056 6 4 6zm0-6c1.6568543 0 3 1.3431457 3 3S5.6568543 6 4 6C2.3431458 6 1 4.6568543 1 3s1.3431458-3 3-3z"
                fill="#3A3B3D"
                fill-rule="evenodd"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              зам
            </p>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
          >
            120
          </h6>
        </div>
      </div>
    </div>
  </div>
</div>
`;
